src/pages/postgraphile/running-postgraphile-in-docker.md

```diff
@@ -0,0 +1,598 @@
+        <td>host: localhost, port: 5432</td>
+        <td>host: your_docker_machine_ip, port: 5432</td>
+    </tr>
+</table>
+
+## Re-initialize The Database
+
+In case you do changes to the database schema by modifying the files in `/db/init`, you will need to re-initialize the database to see these changes. This means you need to delete the Docker Volume, the database Docker Image and rebuild it.
+
+```shell
+# Stop running containers
+$ docker-compose down
+
+# List Docker volumes
+$ docker volume ls
+
+# Delete volume
+$ docker volume rm <your_repository_name>_db
+
+# Delete database image to force rebuild
+$ docker rmi db
+
+# Run containers (will automatically rebuild the image)
+$ docker-compose up
+```
+
+# Add Custom Plugin
+
+## makeWrapResolversPlugin
+
+This section is optional but describes how to wrap a resolver generated by PostGraphile in order to customize it. In the folder `graphql`, create a new subfolder named `custom-plugin`. In this folder create a new file `package.json` with the following content (you can update it to your convenience).
+
+```json
+{
+    "name": "custom-plugin",
+    "version": "0.0.1",
+    "description": "Custom plugin example for PostGraphile.",
+    "main": "index.js",
+    "scripts": {
+        "test": "echo \"Error: no test specified\" && exit 1"
+    },
+    "author": "Alexis ROLLAND",
+    "license": "Apache-2.0",
+    "dependencies": {
+        "graphile-utils": "^4.5.6",
+        "postgraphile": "^4.5.5"
+    }
+}
+```
+
+In the same folder `custom-plugin`, create a new file `index.js` with the following content.
+
+```js
+const { makeWrapResolversPlugin } = require("graphile-utils");
+
+// Create custom wrapper for resolver createUser
+const createUserResolverWrapper = () => {
+    return async (resolve, source, args, context, resolveInfo) => {
+        // You can do something before the resolver executes
+        console.info("Hello world!");
+        console.info(args);
+
+        // Let resolver execute against database
+        const result = await resolve();
+
+        // You can do something after the resolver executes
+        console.info("Hello again!");
+        console.info(result);
+
+        return result;
+    };
+};
+
+// Register custom resolvers
+module.exports = makeWrapResolversPlugin({
+    Mutation: {
+        createUser: createUserResolverWrapper()
+    }
+});
+```
+
+In the `graphql` folder, update the `Dockerfile` so that it looks like the one below.
+
+```dockerfile
+FROM node:alpine
+LABEL description="Instant high-performance GraphQL API for your PostgreSQL database https://github.com/graphile/postgraphile"
+
+# Install PostGraphile and PostGraphile connection filter plugin
+RUN npm install -g postgraphile
+RUN npm install -g postgraphile-plugin-connection-filter
+
+# Install custom plugin
+COPY ./custom-plugin /tmp/custom-plugin
+RUN cd /tmp/custom-plugin && npm pack
+RUN npm install -g /tmp/custom-plugin/custom-plugin-0.0.1.tgz
+RUN rm -rf /tmp/custom-plugin

```

Comment:
Interesting strategy!

Reply:
To be frank I'm not sure how to do that otherwise, I borrowed this piece of code from a colleague.

Reply:
I’d probably have tweaked NODE_PATH personally. Or maybe added `/node_modules` in root.